File: spring-mvc/spring-mvc-1/src/main/java/com/springservletthymeleaf/basic/itemservice/domain/item/ItemResponseMapper.java
package com.springservletthymeleaf.basic.itemservice.domain.item;

import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;

import com.springservletthymeleaf.basic.itemservice.util.GenericMapper;

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface ItemResponseMapper extends GenericMapper<ItemResponse, Item> {
}

File: spring-mvc/spring-mvc-2/src/main/kotlin/com/springthymeleaf/message/ItemResponseMapper.kt
package com.springthymeleaf.message

import org.mapstruct.Mapper
import org.mapstruct.ReportingPolicy

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
interface ItemResponseMapper {

  fun toDto(entity: Item): ItemResponse
  fun toEntity(dto: ItemResponse): Item
  fun toDtoList(entityList: List<Item>?): List<ItemResponse>?
  fun toEntityList(dtoList: List<ItemResponse>?): List<Item>?
}